test/python/test_authentication_external_api.py

```diff
@@ -0,0 +1,205 @@
+            'test',
+            self.account,
+            enabled=True
+        )
+
+        self.assertEqual(status, 204)
+
+    def test_enable_authenticator_instance_401(self):
+        """Test case for enable_authenticator_instance 401 response"""
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.bad_auth_api.enable_authenticator_instance(
+                'oidc',
+                'okta',
+                self.account,
+                enabled=False
+            )
+
+        self.assertEqual(context.exception.status, 401)
+
+    def test_enable_authenticator_instance_404(self):
+        """Test case for enable_authenticator_instance 404 response"""
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.api.enable_authenticator_instance(
+                'oidc',
+                'okta',
+                self.account,
+                enabled=False
+            )
+
+        self.assertEqual(context.exception.status, 404)
+
+    def test_service_login_200(self):
+        """Test case for service_login 200 response
+
+        Login with the given authenticator
+        """
+        alice_config = api_config.get_api_config(username='alice')
+        alice_config.password = 'alice'
+        alice_client = conjur.ApiClient(alice_config)
+        alice_api = conjur.api.AuthenticationApi(alice_client)
+
+        _, status, _ = alice_api.get_api_key_via_ldap_with_http_info('test', self.account)
+
+        self.assertEqual(status, 200)
+
+    def test_service_login_401(self):
+        """Test case for service_login 401 response"""
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.bad_auth_api.get_api_key_via_ldap('aws', self.account)
+
+        self.assertEqual(context.exception.status, 401)
+
+    def test_get_access_token_service_200(self):
+        """Test case for get_access_token_service 200 response
+
+        Login with the given authenticator
+        """
+        alice_config = api_config.get_api_config(username='alice')
+        alice_client = conjur.ApiClient(alice_config)
+        alice_api = conjur.api.AuthenticationApi(alice_client)
+
+        _, status, _ = alice_api.get_access_token_via_ldap_with_http_info(
+            'test',
+            self.account,
+            'alice',
+            body='alice'
+        )
+
+        self.assertEqual(status, 200)
+
+    def test_get_access_token_service_401(self):
+        """Test case for get_access_token_service 401 response
+
+        Login with the given authenticator
+        """
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.api.get_access_token_via_ldap(
+                'test',
+                self.account,
+                'admin',
+                body='test'
+            )
+
+        self.assertEqual(context.exception.status, 401)
+
+# we should write one of these for each service; commenting out for now
```

Comment:
We should remove this comment and open an issue for adding the new tests. We also need to update the old `login` tests to the new `get_api_key`

Reply:
Filed #180